source/slang/slang-serialize-container.h

```diff
@@ -0,0 +1,107 @@
+// slang-serialize-container.h
+#ifndef SLANG_SERIALIZE_CONTAINER_H
+#define SLANG_SERIALIZE_CONTAINER_H
+
+#include "../core/slang-riff.h"
+#include "slang-serialize-types.h"
+#include "slang-ir-insts.h"
+#include "slang-profile.h"
+
+namespace Slang {
+
+class EndToEndCompileRequest;
+
+/* The binary representation actually held in riff/file format*/
+struct SerialContainerBinary
+{
+    struct Target
+    {
+        uint32_t target;
+        uint32_t flags;
+        uint32_t profile;
+        uint32_t floatingPointMode;
+    };
+
+    struct EntryPoint
+    {
+        uint32_t name;
+        uint32_t profile;
+        uint32_t mangledName;
+    };
+};
+
+/* Struct that holds all the data that can be held in a 'container' */
+struct SerialContainerData
+{
+    struct Target
+    {
+        CodeGenTarget       codeGenTarget = CodeGenTarget::Unknown;
+        SlangTargetFlags    flags = 0;
+        Profile             profile;
+        FloatingPointMode   floatingPointMode = FloatingPointMode::Default;
+    };
```

Comment:
It feels like having this type is redundant. We both have the `TargetReq` type that holds the same basic data (and is what we'd likely use this to create) and also the public API's `slang::TargetDesc` type that is nearly identical to this struct except that the fields are in a different order, and there is an extra `optimizationLevel` field not present here.

Is it possible for the (de)serialization logic to directly work with a `slang::TargetDesc` as supported by the public API?